theia-tree-editor/src/browser/tree-editor-widget.tsx

```diff
@@ -79,6 +80,17 @@ export abstract class BaseTreeEditorWidget extends BaseWidget implements Saveabl
     protected init(): void {
         this.configureTitle(this.title);
     }
+    get saveable(): Saveable {
+        return this;
+    }
+
+    createSnapshot(): object {
+        return this.instanceData;
```

Comment:
As this snapshot can be used in any way not known to us and even applied to new widgets (like in copyAndSave), I think we should not just return the object: Editing this snapshot outside the editor would then also edit the data in the editor.

I think we have two solutions:

1. deep clone the object (e.g. by using lodash cloneDeep)
2. Use a text-based snapshot approach. We could `JSON.stringify` the data and return the string. Then we may also be able to implement `applySnapshot` by parsing the given snapshot with `JSON.parse` and setting it to `instanceData`.

I think 2 would be a good approach for now because we don't need an external library and it seems to fulfill the purpose of a snapshot well enough.

What do you think?

Reply:
yes, that would be better. I applied the proposed stringify approach :+1: